spec/models/reservation_spec.rb

```diff
@@ -1,725 +1,913 @@
     end
-    it { is_expected.to be_valid }
   end
 
-  context 'when user has overdue reservation' do
-    subject(:reservation) { FactoryGirl.build(:valid_reservation) }
-    let(:overdue_reserver) { reservation.reserver }
-    let!(:overdue) do
-      o = FactoryGirl.build(:overdue_reservation, reserver: overdue_reserver)
-      o.save(validate: false)
-      o
+  describe '#overlaps_with' do
+    let!(:res) do
+      FactoryGirl.build_stubbed(:valid_reservation,
+                                start_date: Time.zone.today,
+                                due_date: Time.zone.today + 1.day)
     end
-
-    it { is_expected.to be_valid }
-    it 'should not save' do
-      expect(reservation.save).to be_truthy
-      expect(Reservation.all.size).to eq(2)
-      expect(Reservation.all.first).to eq(overdue)
-    end
-    it 'can be updated' do
-      reservation.start_date = Time.zone.today + 1.day
-      expect(reservation.save).to be_truthy
-    end
-    it 'fails appropriate validations' do
-      expect(reservation.validate).not_to eq([])
-    end
-    it 'passes other custom validations' do
-      expect(reservation.start_date_before_due_date).to be_nil
-      expect(reservation.not_empty).to be_nil
-      expect(reservation.matched_item_and_model).to be_nil
-      expect(reservation.available).to be_nil
-      expect(reservation.not_in_past).to be_nil
-      expect(reservation.check_banned).to be_nil
+    it 'returns true when overlapping with date' do
+      expect(res.overlaps_with(Time.zone.today)).to be_truthy
+    end
+    it 'returns false when not overlapping with date' do
+      expect(res.overlaps_with(Time.zone.today - 1.day)).to be_falsey
     end
   end
 
-  context 'with banned user' do
-    let(:banned) { FactoryGirl.create(:banned) }
-    subject(:reservation) do
-      FactoryGirl.build(:valid_reservation, reserver_id: banned.id)
+  describe '#flag' do
+    let!(:res) { FactoryGirl.build_stubbed(:valid_reservation) }
+    it 'flags the reservation' do
+      expect { res.flag(:request) }.to \
+        change { res.flagged?(:request) }.from(false).to(true)
     end
-
-    it { is_expected.not_to be_valid }
-    it 'should not save' do
-      expect(reservation.save).to be_falsey
+    it 'does nothing if flag is undefined' do
+      expect { res.flag(:garbage) }.not_to change { res.flags }
     end
-    it 'fails appropriate validations' do
-      expect(reservation.validate).not_to eq([])
+    it 'does nothing if flag is already set' do
+      res.flag(:request)
+      expect { res.flag(:request) }.not_to change { res.flags }
     end
-    it 'passes other custom validations' do
-      expect(reservation.start_date_before_due_date).to be_nil
-      expect(reservation.not_empty).to be_nil
-      expect(reservation.matched_item_and_model).to be_nil
-      expect(reservation.available).to be_nil
-      expect(reservation.not_in_past).to be_nil
+    it "doesn't save the reservation" do
```

Comment:
Why do we need to test this? I feel like you generally don't test messages that *aren't* sent...